src/Build/Definition/ProjectCollection.cs

```diff
@@ -497,6 +500,11 @@ public static string DisplayVersion
             }
         }
 
+        /// <summary>
+        /// Properties passed from the command line (e.g. by using /p:).
+        /// </summary>
+        public HashSet<string> PropertiesFromCommandLine { get; }
```

Comment:
I think `ICollection<T>` should be sufficient since we only call `.Contains`

Always limit the interface to the minimal API surface you're actually going to need.

Here I'd perhaps even consider `Func<string> IsPropertyFromCommandLine` and pass around a delegate. But may be less intuitive than `ICollection<T>` which seems just right.